sklearn/datasets/openml.py

```diff
@@ -0,0 +1,200 @@
+
+    Datasets are uniquely identified by either an integer ID or by a
+    combination of name and version (i.e. there might be multiple
+    versions of the 'iris' dataset).
+
+    Parameters
+    ----------
+    name_or_id : string or integer
+        Identifier of the dataset. If integer, assumed to be the id of the
+        dataset on OpenML, if string, assumed to be the name of the dataset.
+
+    version : integer or 'active', default='active'
+        Version of the dataset. Only used if ``name_or_id`` is a string.
+        If 'active' the oldest version that's still active is used.
+
+    data_home : string or None, default None
+        Specify another download and cache folder for the data sets. By default
+        all scikit-learn data is stored in '~/scikit_learn_data' subfolders.
+
+    target_column : string or None, default 'default-target'
+        Specify the column name in the data to use as target. If
+        'default-target', the standard target column a stored on the server
+        is used. If ``None``, all columns are returned as data and the
+        tharget is ``None``.
+
+    memory : boolean, default=True
+        Whether to store downloaded datasets using joblib.
+
+    Returns
+    -------
+
+    data : Bunch
+        Dictionary-like object, the interesting attributes are:
+        'data', the data to learn, 'target', the regression target or
+        classification labels, 'DESCR', the full description of the dataset,
+        'feature_names', the original names of the dataset columns, and
+        'details' which provide more information on the openml meta-data.
+    """
+    data_home = get_data_home(data_home=data_home)
+    data_home = join(data_home, 'openml')
+    if memory:
+        mem = Memory(join(data_home, 'cache'), verbose=0).cache
+    else:
+        def mem(func):
+            return func
+    _get_data_info_by_name_ = mem(_get_data_info_by_name)
+    _get_data_description_by_id_ = mem(_get_data_description_by_id)
+    _download_data_ = mem(_download_data)
+
+    if not exists(data_home):
+        os.makedirs(data_home)
+
+    # check if dataset id is known
+    if isinstance(name_or_id, numbers.Integral):
+        if version != "active":
+            raise ValueError(
+                "Dataset id={} and version={} passed, but you can only "
+                "specify a numeric id or a version, not both.".format(
+                    name_or_id, version))
+        data_id = name_or_id
+    elif isinstance(name_or_id, str):
+        data_info = _get_data_info_by_name_(name_or_id, version)
+        data_id = data_info['did']
+
+    else:
+        raise TypeError(
+            "Invalid name_or_id {}, should be string or integer.".format(
+                name_or_id))
+
+    data_description = _get_data_description_by_id_(data_id)
+    if data_description['status'] != "active":
+        warn("Version {} of dataset {} is inactive, meaning that issues have"
```

Comment:
perhaps we need to check that `data_description['status'] == 'inactive'` and raise a different warning if it's not, just in case there are different statuses....

Reply:
there's also "in review". I added some docs here: https://github.com/openml/website/pull/210